Analysis/SLDCorrection/include/SLDCorrection.h

```diff
@@ -0,0 +1,598 @@
+	DoubleVector				m_sigmaUzUz{};
+	DoubleVector				m_normalizedResidualFlightDirectionUx{};
+	DoubleVector				m_normalizedResidualFlightDirectionUy{};
+	DoubleVector				m_normalizedResidualFlightDirectionUz{};
+	DoubleVector				m_trueNeutrinoFourMomentumPx{};
+	DoubleVector				m_trueNeutrinoFourMomentumPy{};
+	DoubleVector				m_trueNeutrinoFourMomentumPz{};
+	DoubleVector				m_trueNeutrinoFourMomentumE{};
+	DoubleVector				m_recoNeutrinoFourMomentumClosePx{};
+	DoubleVector				m_recoNeutrinoFourMomentumClosePy{};
+	DoubleVector				m_recoNeutrinoFourMomentumClosePz{};
+	DoubleVector				m_recoNeutrinoFourMomentumCloseE{};
+	DoubleVector				m_sigmaNeutrinoPxPx{};
+	DoubleVector				m_sigmaNeutrinoPxPy{};
+	DoubleVector				m_sigmaNeutrinoPyPy{};
+	DoubleVector				m_sigmaNeutrinoPxPz{};
+	DoubleVector				m_sigmaNeutrinoPyPz{};
+	DoubleVector				m_sigmaNeutrinoPzPz{};
+	DoubleVector				m_sigmaNeutrinoPxE{};
+	DoubleVector				m_sigmaNeutrinoPyE{};
+	DoubleVector				m_sigmaNeutrinoPzE{};
+	DoubleVector				m_sigmaNeutrinoEE{};
+	DoubleVector				m_residualNeutrinoFourMomentumPx{};
+	DoubleVector				m_residualNeutrinoFourMomentumPy{};
+	DoubleVector				m_residualNeutrinoFourMomentumPz{};
+	DoubleVector				m_residualNeutrinoFourMomentumE{};
+	DoubleVector				m_normalizedResidualNeutrinoFourMomentumPx{};
+	DoubleVector				m_normalizedResidualNeutrinoFourMomentumPy{};
+	DoubleVector				m_normalizedResidualNeutrinoFourMomentumPz{};
+	DoubleVector				m_normalizedResidualNeutrinoFourMomentumE{};
+	DoubleVector				m_recoNeutrinoDirectionError{};
+	DoubleVector				m_PCAatLeptonX{};
+	DoubleVector				m_PCAatLeptonY{};
+	DoubleVector				m_PCAatLeptonZ{};
+	DoubleVector				m_PCAatOtherParticleX{};
+	DoubleVector				m_PCAatOtherParticleY{};
+	DoubleVector				m_PCAatOtherParticleZ{};
+	DoubleVector				m_JetAxisX{};
+	DoubleVector				m_JetAxisY{};
+	DoubleVector				m_JetAxisZ{};
+	DoubleVector				m_PrimaryVertexX{};
+	DoubleVector				m_PrimaryVertexY{};
+	DoubleVector				m_PrimaryVertexZ{};
+	DoubleVector				m_AngleTrueFlightDirectionJet{};
+	DoubleVector				m_AngleRecoFlightDirectionJet{};
+	DoubleVector				m_AngleLeptonJet{};
+	DoubleVector				m_AngleDSVertexJet{};
+	DoubleVector				m_AngleRecoNeutrinoJet{};
+	DoubleVector				m_AngleTrueNeutrinoJet{};
+	DoubleVector				m_LeptonDistanceFromPV{};
+	DoubleVector				m_DSVDistanceFromPV{};
+	DoubleVector				m_Lepton3DImpactParameter{};
+	DoubleVector				m_OtherParticle3DImpactParameter{};
+	IntVector				m_SLDStatus{};
+	IntVector				m_PVAStatus{};
+	IntVector				m_trueSolutionSign{};
+	IntVector				m_recoSolutionSign{};
+	IntVector				m_nChargedParticlesInPrimVertex{};
+	IntVector				m_nChargedParticlesNotInPrimVertex{};
+	FloatVector				m_weightPFOtoMCP_Lepton{};
+	FloatVector				m_weightMCPtoPFO_Lepton{};
+	FloatVector				m_weightPFOtoMCP_Neutral{};
+	FloatVector				m_weightMCPtoPFO_Neutral{};
+	FloatVector				m_weightPFOtoMCP_Charged{};
+	FloatVector				m_weightMCPtoPFO_Charged{};
+	TH1I					*h_SLDStatus{};
```

Comment:
All of these histograms seem to be leaked (or at least I can't see where they are cleaned up). I would suggest making them `std::unique_ptr` instead of raw pointers and then use `std::make_unique` instead of `new` to create them. This way they are automatically cleaned up. E.g.

```cpp
// member declaration in the header
std::unique_ptr<TH1I> h_SLDStatus{nullptr};

// initialization in the source file
h_SLDStatus = std::make_unique<TH1I>("SLDStatus" , ";" , 7 , 0 , 7);
```